src/Tuura/Concept/Circuit/Derived.hs

```diff
@@ -89,41 +89,53 @@ dual c = mempty
          }
 
 -- Give the opposite initial state for the chosen signal
-bubbleInitialValue :: Eq a => a -> (a -> InitialValue) -> (a -> InitialValue)
-bubbleInitialValue s f y = if y == s then bubbleVal (f y) else f y
+bubbleInitialValue :: Eq a => [a] -> (a -> InitialValue) -> (a -> InitialValue)
+bubbleInitialValue s f y = if y `elem` s then bubbleVal (f y) else f y
   where
     bubbleVal (Defined v) = Defined (not v)
     bubbleVal x = x
 
 -- Perform the inversion for just transitions of the given signal
-toggleSpecific :: Eq a => a -> Transition a -> Transition a
+toggleSpecific :: Eq a => [a] -> Transition a -> Transition a
 toggleSpecific a t
-    | signal t == a = toggle t
-    | otherwise     = t
+    | signal t `elem` a = toggle t
+    | otherwise         = t
 
 -- For the selected signal, toggle the directions of all transitions for this.
-bubbleCausality :: Eq a => a -> Causality (Transition a)
+bubbleCausality :: Eq a => [a] -> Causality (Transition a)
                         -> Causality (Transition a)
```

Comment:
Don't break this type signature, it's OK to occasionally exceed the line width limit.